ShareX/CaptureHelpers/CaptureRegion.cs

```diff
@@ -87,6 +88,47 @@ protected ImageInfo ExecuteRegionCapture(TaskSettings taskSettings)
 
             using (RegionCaptureForm form = new RegionCaptureForm(mode, taskSettings.CaptureSettingsReference.SurfaceOptions, bmp))
             {
+                form.SaveImageRequested += (output, newFilePath) =>
+                {
+                    using (output)
+                    {
+                        if (string.IsNullOrEmpty(newFilePath))
+                        {
+                            string fileName = TaskHelpers.GetFilename(taskSettings, taskSettings.ImageSettings.ImageFormat.GetDescription(), output);
+                            newFilePath = Path.Combine(taskSettings.GetScreenshotsFolder(), fileName);
+                        }
+
+                        ImageHelpers.SaveImage(output, newFilePath);
+                    }
+
+                    return newFilePath;
+                };
+
+                form.SaveImageAsRequested += (output, newFilePath) =>
+                {
+                    using (output)
+                    {
+                        if (string.IsNullOrEmpty(newFilePath))
+                        {
+                            string fileName = TaskHelpers.GetFilename(taskSettings, taskSettings.ImageSettings.ImageFormat.GetDescription(), output);
+                            newFilePath = Path.Combine(taskSettings.GetScreenshotsFolder(), fileName);
+                        }
+
+                        newFilePath = ImageHelpers.SaveImageFileDialog(output, newFilePath);
```

Comment:
the difference in one line. could it be parametrized?

Reply:
it's just copied from method AnnotateImage in file TaskHelpers.cs https://github.com/ShareX/ShareX/blob/a883f2bf63aaf5a30182fc61299a8d9da3087064/ShareX/TaskHelpers.cs#L960-L998